.changes/nextrelease/mockhandlerExcpetionFix.json

```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "type": "enhancement",
+    "category": "",
+    "description": "Code changes to mock exceptions correctly in mockhandler"
+  }
+]
```

Comment:
Not gonna block approval of this, but worth noting: see the circle with the red line at the end of this file?  That's because PSR-2 guidelines recommending files ending with a blank line (https://www.php-fig.org/psr/psr-2/#22-files).  This file is going to get automatically deleted and the contents moved to a changelog file anyway though, so being nit-picky about this one isn't necessary